dashboard/app/controllers/activities_controller.rb

```diff
@@ -160,16 +162,27 @@ def track_progress_for_user
     else
       @activity = Activity.create_async!(attributes)
     end
-
     if @script_level
-      @new_level_completed = current_user.track_level_progress_async(
-        script_level: @script_level,
-        new_result: test_result,
-        submitted: params[:submitted] == "true",
-        level_source_id: @level_source.try(:id),
-        level: @level,
-        pairing_user_ids: pairing_user_ids
-      )
+      if synchronous_save
```

Comment:
Yes, there is a particular motivation, related only to the changes made in this PR. In fact, I'd encourage you understand this motivation and its implications before LGTM'ing my change. In the async case, we are not assured the associated `UserLevel` will exist when the `GalleryActivity` is created below. Thus, anytime we might create the `GalleryActivity`, we need to synchronously create the UserLevel, so that we can use it to populate the `GalleryActivity`. But if we know we aren't creating a `GalleryActivity`, we can respect the GateKeeper settings to create the `UserLevel` synchronously or asynchronously.